core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -127,24 +135,39 @@ public static MetricsConfig forPositionDelete(Table table) {
     return new MetricsConfig(columnModes.build(), defaultMode);
   }
 
-  private static MetricsConfig from(Map<String, String> props, SortOrder order) {
+  /**
+   * Generate a MetricsConfig for all columns based on overrides, sortOrder, and defaultMode.
+   * @param props will be read for metrics overrides (write.metadata.metrics.column.*) and default
+   *              (write.metadata.metrics.default)
+   * @param order sort order columns, will be promoted to truncate(16)
+   * @param defaultMode MetricsConfig default, if not set by user property
+   * @return metrics configuration
+   */
+  private static MetricsConfig from(Map<String, String> props, SortOrder order, MetricsMode defaultMode) {
     Map<String, MetricsMode> columnModes = Maps.newHashMap();
-    MetricsMode defaultMode;
-    String defaultModeAsString = props.getOrDefault(DEFAULT_WRITE_METRICS_MODE, DEFAULT_WRITE_METRICS_MODE_DEFAULT);
-    try {
-      defaultMode = MetricsModes.fromString(defaultModeAsString);
-    } catch (IllegalArgumentException err) {
-      // Mode was invalid, log the error and use the default
-      LOG.warn("Ignoring invalid default metrics mode: {}", defaultModeAsString, err);
-      defaultMode = MetricsModes.fromString(DEFAULT_WRITE_METRICS_MODE_DEFAULT);
+
+    // Handle user override of default mode
+    MetricsMode finalDefaultMode;
+    String defaultModeProp = props.get(DEFAULT_WRITE_METRICS_MODE);
+    if (defaultModeProp != null) {
+      try {
+        finalDefaultMode = MetricsModes.fromString(defaultModeProp);
+      } catch (IllegalArgumentException err) {
+        // User override was invalid, log the error and use the default
+        LOG.warn("Ignoring invalid default metrics mode: {}", defaultModeProp, err);
```

Comment:
I'm not sure why we had this as a log warn before, seems like this should just fail ... but I don't think we need to change the behavior yet

Reply:
I might be mistaken, but I think it was the same behavior, warn log is still there. but in any case after changing the default from enum back to a string the diff now looks less.  Now in this method only the original variable name "defaultMode" should be changed to avoid conflict.

Reply:
Yeah I just don't know why we originally had this behavior ... seems odd to me :shrug:

Reply:
The idea here was to not fail writes just because a user messed up a tangentially related config property. I think it's better to allow pipelines to continue.